modules/apps/29-fee/keeper/escrow.go

```diff
@@ -26,64 +26,63 @@ func (k Keeper) EscrowPacketFee(ctx sdk.Context, identifiedFee types.IdentifiedP
 		return sdkerrors.Wrapf(types.ErrRefundAccNotFound, "account with address: %s not found", refundAcc)
 	}
 
-	coins := identifiedFee.Fee.RecvFee
-	coins = coins.Add(identifiedFee.Fee.AckFee...)
-	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
-
-	if err := k.bankKeeper.SendCoinsFromAccountToModule(
-		ctx, refundAcc, types.ModuleName, coins,
-	); err != nil {
+	coins := identifiedFee.Fee.Total()
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, refundAcc, types.ModuleName, coins); err != nil {
 		return err
 	}
 
-	// Store fee in state for reference later
-	k.SetFeeInEscrow(ctx, identifiedFee)
+	packetFees := []types.IdentifiedPacketFee{identifiedFee}
+	if feesInEscrow, found := k.GetFeesInEscrow(ctx, identifiedFee.PacketId); found {
+		packetFees = append(packetFees, feesInEscrow.PacketFees...)
+	}
+
+	identifiedFees := types.NewIdentifiedPacketFees(packetFees)
+	k.SetFeesInEscrow(ctx, identifiedFee.PacketId, identifiedFees)
+
 	return nil
 }
 
 // DistributePacketFees pays the acknowledgement fee & receive fee for a given packetId while refunding the timeout fee to the refund account associated with the Fee.
-func (k Keeper) DistributePacketFees(ctx sdk.Context, refundAcc, forwardRelayer string, reverseRelayer sdk.AccAddress, feeInEscrow types.IdentifiedPacketFee) {
-	// distribute fee for forward relaying
-	forward, err := sdk.AccAddressFromBech32(forwardRelayer)
-	if err == nil {
-		k.distributeFee(ctx, forward, feeInEscrow.Fee.RecvFee)
-	}
-
-	// distribute fee for reverse relaying
-	k.distributeFee(ctx, reverseRelayer, feeInEscrow.Fee.AckFee)
+func (k Keeper) DistributePacketFees(ctx sdk.Context, forwardRelayer string, reverseRelayer sdk.AccAddress, feesInEscrow []types.IdentifiedPacketFee) {
+	for _, packetFee := range feesInEscrow {
+		// distribute fee for forward relaying
+		forward, err := sdk.AccAddressFromBech32(forwardRelayer)
+		if err == nil {
+			k.distributeFee(ctx, forward, packetFee.Fee.RecvFee)
+		}
 
-	// refund timeout fee refund,
-	refundAddr, err := sdk.AccAddressFromBech32(refundAcc)
-	if err != nil {
-		panic(fmt.Sprintf("could not parse refundAcc %s to sdk.AccAddress", refundAcc))
-	}
+		// distribute fee for reverse relaying
+		k.distributeFee(ctx, reverseRelayer, packetFee.Fee.AckFee)
 
-	// refund timeout fee for unused timeout
-	k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.TimeoutFee)
+		// refund timeout fee refund,
+		refundAddr, err := sdk.AccAddressFromBech32(packetFee.RefundAddress)
```

Comment:
Pull line outside for loop

Reply:
This needs to be inside the loop. The `RefundAddress` exists as a field on the `packetFee` and it may be different for additional fees

Reply:
Ahh correct